### PR TITLE
Add missing Realm.deleteRealmFile(File) method

### PR DESCRIPTION
**Pull request by cmelchior**
Fixes https://github.com/realm/realm-java/issues/991

@emanuelez @kneth @bmunkholm

### CODE REVIEW

File: realm/src/main/java/io/realm/Realm.java [hunk 1]
```diff
@@ -1682,20 +1682,30 @@ public static boolean deleteRealmFile(Context context) {
...
      * @return false if a file could not be deleted. The failing file will be logged.
      */
-    public static synchronized boolean deleteRealmFile(Context context, String fileName) {
+    public static boolean deleteRealmFile(Context context, String fileName) {
+        return deleteRealmFile(new File(context.getFilesDir(), fileName));
+    }
+
+    /**
+     * Delete the Realm file from the filesystem for a custom named Realm.
+     * The realm must be unused and closed before calling this method.
```
Comment (kneth) [inline realm/src/main/java/io/realm/Realm.java]: "realm" -> "Realm"

File: realm/src/main/java/io/realm/Realm.java [hunk 2]
```diff
@@ -1682,20 +1682,30 @@ public static boolean deleteRealmFile(Context context) {
...
         int realmId = realmFile.getAbsolutePath().hashCode();
-
         AtomicInteger counter = openRealms.get(realmId);
         if (counter != null && counter.get() > 0) {
             throw new IllegalStateException("It's not allowed to delete the file associated with an open Realm. " +
                     "Remember to close() all the instances of the Realm before deleting its file.");
         }
 
-        List<File> filesToDelete = Arrays.asList(realmFile, new File(writableFolder, fileName + ".lock"));
+        List<File> filesToDelete = Arrays.asList(realmFile, new File(realmFolder, fileName + ".lock"));
```
Comment (kneth) [inline realm/src/main/java/io/realm/Realm.java]: Should we delete the `.log_a` and `.log_b` files?

**Comment by cmelchior** [comment]
Yes, we should. I guess this was changed after core updated the lock file mechanism.

**Comment by kneth** [comment]
:+1:

File: realm/src/androidTest/java/io/realm/RealmTest.java [hunk 3]
```diff
@@ -1529,4 +1529,12 @@ public void testUpdateObjectWithLinks() throws Exception {
 
         testRealm.commitTransaction();
     }
+
+    public void testDeleteNonRealmFile() throws IOException {
+        File tmpFile = new File(getContext().getFilesDir(), "tmp");
+        tmpFile.delete();
+        assertTrue(tmpFile.createNewFile());
+        assertTrue(Realm.deleteRealmFile(tmpFile));
+        tmpFile.delete();
```
Comment (emanuelez) [inline realm/src/androidTest/java/io/realm/RealmTest.java]: why deleting the tmpfile here?

**Comment by cmelchior** [comment]
Fixed @emanuelez

**Comment by emanuelez** [comment]
:+1:

**Pull request merged**